packages/react/src/components/badge/Badge.tsx

```diff
@@ -24,35 +63,39 @@ interface BadgeWithIconProps extends BadgeBasicProps {
      * Add an icon to the badge (Required if no label)
      */
     icon: SvgName;
+    /**
+     * Whether the icon is left or right (Required if no label)
+     */
+    iconPlacement: BadgeIconPlacement;
 }
-
 export type IBadgeProps = BadgeWithLabelProps | BadgeWithIconProps | (BadgeWithLabelProps & BadgeWithIconProps);
 
 export class Badge extends React.Component<IBadgeProps> {
-    static defaultProps: Partial<IBadgeProps> = {
+    static defaultProps: IBadgeProps = {
         extraClasses: [],
+        type: BadgeType.Default,
+        isSmall: false,
+        icon: undefined,
+        iconPlacement: undefined,
     };
 
-    private get isSmall(): boolean {
-        return this.className.includes('mod-small');
-    }
-
     private get className(): string {
-        return classNames(DEFAULT_BADGE_CLASSNAME, this.props.extraClasses);
+        return classNames(DEFAULT_BADGE_CLASSNAME, TypeClassMapping[this.props.type], this.props.extraClasses, {
+            'mod-small': this.props.isSmall,
+        });
     }
 
     render() {
         return (
-            <span className={this.className} aria-label="badge">
-                {'icon' in this.props ? (
-                    <Svg
-                        svgName={this.props.icon}
-                        svgClass={classNames('icon', {'mod-16': !this.isSmall, 'mod-12': this.isSmall})}
-                        className={classNames({mr1: 'label' in this.props && this.props.label})}
-                    />
+            <div className={this.className} aria-label="badge">
+                {'icon' in this.props && this.props.icon && IconPlacementMapping[this.props.iconPlacement] ? (
+                    <Svg svgName={this.props.icon} className={'mod-badge'} svgClass={'icon mod-badge'} />
+                ) : null}

```

Comment:
I'm thinking that we should move the icon with CSS instead of having it render at a different place. Since you're using flex you can check the `order` property. (I think it would be better to always have it after the label)

```suggestion
                {'icon' in this.props ? (
                    <Svg svgName={this.props.icon} className={className('mod-badge', {'mod-right': this.props.iconPlacement === BadgeIconPlacement.Right})} svgClass={'icon mod-badge'} />
                ) : null}
```